### ISSUE TITLE
Bug: sam remote invoke - ModuleNotFoundError

### ISSUE THREAD

**Issue by cm-suzuki-junya** [aws/aws-sam-cli#5566]
### Description:
execute `sam remote invoke`.

an error occurred `Error: No module named 'mypy_boto3_lambda'`

### Steps to reproduce:
1. Execute `sam init`
2. Execute `sam deploy`
3. Execute `sam remote invoke`

sam init message result
```
    -----------------------
    Generating application:
    -----------------------
    Name: sam-app-remote
    Runtime: python3.9
    Architectures: x86_64
    Dependency Manager: pip
    Application Template: hello-world
    Output Directory: .
    Configuration file: sam-app-remote/samconfig.toml
```

### Observed result:
```
$ sam remote invoke --debug
2023-07-19 19:04:36,898 | Config file location: /xxxx/sam-app-remote/samconfig.toml
2023-07-19 19:04:36,899 | Loading configuration values from [default.['remote', 'invoke'].parameters] (env.command_name.section) in config file at '/xxxx/sam-app-remote/samconfig.toml'...
2023-07-19 19:04:36,900 | Configuration values successfully loaded.
2023-07-19 19:04:36,900 | Configuration values are: {'stack_name': 'sam-app-remote'}
2023-07-19 19:04:36,900 | Converting provided --output option value to Enum
2023-07-19 19:04:36,935 | Using config file: samconfig.toml, config environment: default
2023-07-19 19:04:36,936 | Expand command line arguments to:
2023-07-19 19:04:36,936 | --stack_name=sam-app-remote --output=RemoteInvokeOutputFormat.TEXT
2023-07-19 19:04:36,937 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics
2023-07-19 19:04:36,984 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics
2023-07-19 19:04:36,986 | Unable to find Click Context for getting session_id.
2023-07-19 19:04:36,986 | Sending Telemetry: {'metrics': [{'commandRun': {'requestId': '467cc95e-b86a-4fa8-ac84-1afb86a1556e', 'installationId': '4e91b0f1-0a87-4da7-95f9-c9c6301a0dc0', 'sessionId': 'a02ab33d-bd73-4527-83ed-02333e1f4cf2',
'executionEnvironment': 'CLI', 'ci': False, 'pyversion': '3.11.4', 'samcliVersion': '1.91.0', 'awsProfileProvided': False, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam remote invoke', 'metricSpecificAttributes':
{'projectType': 'CFN', 'gitOrigin': None, 'projectName': '8ecbd3c825736f3f328990f07cd180720320559db28602c0b469ea1f3a07bf20', 'initialCommit': None}, 'duration': 2, 'exitReason': 'ModuleNotFoundError', 'exitCode': 255}}]}
2023-07-19 19:04:36,988 | Sending Telemetry: {'metrics': [{'events': {'requestId': '0a143a3b-3302-48f2-88e2-fa5f3e39324b', 'installationId': '4e91b0f1-0a87-4da7-95f9-c9c6301a0dc0', 'sessionId': 'a02ab33d-bd73-4527-83ed-02333e1f4cf2',
'executionEnvironment': 'CLI', 'ci': False, 'pyversion': '3.11.4', 'samcliVersion': '1.91.0', 'metricSpecificAttributes': {'events': [{'event_name': 'SamConfigFileExtension', 'event_value': '.toml', 'thread_id':
'68cac41a737f4a7b9c50bcc9a948655e', 'time_stamp': '2023-07-19 10:04:36.898', 'exception_name': None}]}}}]}
2023-07-19 19:04:37,513 | HTTPSConnectionPool(host='aws-serverless-tools-telemetry.us-west-2.amazonaws.com', port=443): Read timed out. (read timeout=0.1)
2023-07-19 19:04:37,516 | HTTPSConnectionPool(host='aws-serverless-tools-telemetry.us-west-2.amazonaws.com', port=443): Read timed out. (read timeout=0.1)

Error: No module named 'mypy_boto3_lambda'
Traceback:
  File "/private/tmp/sam191/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/lib/cli_validation/remote_invoke_options_validations.py", line 89, in wrapped
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/lib/cli_validation/remote_invoke_options_validations.py", line 52, in wrapped
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/lib/telemetry/metric.py", line 184, in wrapped
    raise exception  # pylint: disable=raising-bad-type
    ^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/lib/telemetry/metric.py", line 149, in wrapped
    return_value = func(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/lib/utils/version_checker.py", line 42, in wrapped
    actual_result = func(*args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/cli/main.py", line 95, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/commands/_utils/command_exception_handler.py", line 88, in wrapper_command_exception_handler
    raise ex
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/commands/_utils/command_exception_handler.py", line 68, in wrapper_command_exception_handler
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/commands/remote/invoke/cli.py", line 91, in cli
    do_cli(
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/commands/remote/invoke/cli.py", line 121, in do_cli
    from samcli.commands.remote.remote_invoke_context import RemoteInvokeContext
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/commands/remote/remote_invoke_context.py", line 18, in <module>
    from samcli.lib.remote_invoke.remote_invoke_executor_factory import RemoteInvokeExecutorFactory
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/lib/remote_invoke/remote_invoke_executor_factory.py", line 7, in <module>
    from samcli.lib.remote_invoke.lambda_invoke_executors import (
  File "/private/tmp/sam191/lib/python3.11/site-packages/samcli/lib/remote_invoke/lambda_invoke_executors.py", line 14, in <module>
    from mypy_boto3_lambda.client import LambdaClient
```

### Expected result:

Lambda function on remote is executed

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: macOS 13.4
4. `sam --version`: `SAM CLI, version 1.91.0`
5. AWS region: ap-northeast-1

```
$ sam --info
{
  "version": "1.91.0",
  "system": {
    "python": "3.11.4",
    "os": "macOS-13.4-arm64-arm-64bit"
  },
  "additional_dependencies": {
    "docker_engine": "20.10.17",
    "aws_cdk": "Not available",
    "terraform": "1.3.2"
  },
  "available_beta_feature_env_vars": [
    "SAM_CLI_BETA_FEATURES",
    "SAM_CLI_BETA_BUILD_PERFORMANCE",
    "SAM_CLI_BETA_TERRAFORM_SUPPORT",
    "SAM_CLI_BETA_RUST_CARGO_LAMBDA"
  ]
}
```

## Resolve

add `mypy-boto3-lambda`

```
pip install mypy-boto3-lambda -t /tmp/sam191/lib/python3.11/site-packages
```

A similar problem occurs for StepFunctions.

```
pip install mypy-boto3-stepfunctions -t /tmp/sam191/lib/python3.11/site-packages
```

NOTE: SAM is installed in the environment created by `python -m venv /tmp/sam191`.

**Comment by mildaniel** [comment]
Hi @cm-suzuki-junya, thanks for opening this issue. We have confirmed that this is a bug with the latest version of AWS SAM CLI and are working on a fix. Thanks for providing a workaround for those who install using pip.

**Comment by mildaniel** [comment]
A fix has been merged in [version 1.92.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.92.0)! Please open a new issue if you find any other problems.
